modules/mqtt/destination/mqtt-worker.c

```diff
@@ -183,6 +183,7 @@ _thread_init(LogThreadedDestWorker *s)
   MQTTDestinationDriver *owner = (MQTTDestinationDriver *) s->owner;
 
   g_string_assign(self->topic, owner->topic->str);
+  self->qos = owner->qos;
```

Comment:
`qos` is the same for all the workers. It is enough to store it in the driver, and use `self->owner->qos` everywhere.